.github/workflows/test.yaml

```diff
@@ -110,7 +119,7 @@ jobs:
         tox ${{ matrix.trial-target }}
 
     - name: Prepare coverage
-      if: always() && contains(matrix['tox-env'], 'withcov')
+      if: ${{ !cancelled() && contains(matrix['tox-env'], 'withcov') }}
```

Comment:
I'm always uncomfortable with the GitHub documentation's wording around `if:` and the conditions.  I thought `always()` was some weird thing and that without it you didn't get the run-even-if-a-previous-step-failed behavior.  But, I might be making this up so if you think it's OK, don't worry too much.  Just noting for completeness.